ivy/stateful/norms.py

```diff
@@ -175,7 +177,7 @@ def _create_variables(self, device, dtype=None):
     def _forward(
         self,
         inputs,
-        training: bool = False,
+        training: bool = None,
```

Comment:
let's remove this argument, I think we might just introduce confusion by adding an additional argument where `model.train()` and `model.eval()` should really be the only way to control the training mode passed. So provided we ensure that, only passing `self.training` on line [206](https://github.com/unifyai/ivy/pull/22469/files#diff-08ded1880c50b8c039f86533aab4b2b05be766fe26e9c7edc8d4655f4acb6149R206) without any additional handling should work

Reply:
but how would this be different from doing a `model.train()` or `model.eval()` on a BatchNorm module. If we consider the `torch` design for e.g., [`torch.nn.functional.batch_norm`](https://pytorch.org/docs/stable/generated/torch.nn.functional.batch_norm.html) has a `training` parameter, but [`torch.nn.BatchNorm2d`](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html#torch.nn.BatchNorm2d) doesn't have such a parameter, and it doesn't need it either given that there's `model.train()` and `model.eval()` 😅
Secondly it becomes hard to debug if this `training` parameter overwrites `self.training` because we already have `model.train()` and `model.eval()` to do that.
To your point of being able to control this to the module, we could accept `training` as an argument to the `ivy.Module.__init__` which would be `True` by default, and then once a `ivy.Module` is created, the only way to modify it would be by calling `model.train()` or `model.eval()` (for the parent module or any child module) 😄

Reply:
In that case, we would need to get rid of the argument from the _forward method of this implementation.